File: LeetCode_Solutions/3291-find-if-array-can-be-sorted/find-if-array-can-be-sorted.java
class Solution {
    public boolean canSortArray(int[] nums) {
        int bitcnt=Integer.bitCount(nums[0]);
        int prev=-1;
        int cur=nums[0];
        for(int i=1;i<nums.length;i++){
            if(Integer.bitCount(nums[i])!=bitcnt){
                bitcnt=Integer.bitCount(nums[i]);
                prev=Math.max(cur,prev);
                cur=nums[i];
            }
            else{
                cur=Math.max(cur,nums[i]);
            }
            if(nums[i]<prev){
                return false;
            }
        }
        return true;
    }
}

File: LeetCode_Solutions/3291-find-if-array-can-be-sorted/find-if-array-can-be-sorted.py
class Solution:
    def canSortArray(self, nums: List[int]) -> bool:
        bitcnt=bin(nums[0]).count('1')
        prev=-1
        cur=nums[0]
        for i in range(1,len(nums)):
            if bin(nums[i]).count('1')!=bitcnt:
                bitcnt=bin(nums[i]).count('1')
                prev=max(prev,cur)
                cur=nums[i]
            else:
                cur=max(nums[i],cur)
            if prev>nums[i]:
                return False
        return True